Tune the writeup

NicMcPhee opened the pull request:
Various changes to README and LABTASKS, mostly aimed at improving clarity.

NicMcPhee committed:
Update README.md

A variety of edits, mostly aimed at trying to clarify our intentions or language.

* I brought the role of Zen Garden in this more up to date.
* I said more about responsive design (which had really been relegated to the LABTASKS document.
* Added some resources on good commit messages and pull requests.
* Removed the "CSS3 Features" link because it was dead (and way out of date)
* Removed the "Daily CSS Fun" link as that hadn't been updated in nearly 20 years.
* Added another Flexbox resource link

NicMcPhee committed:
Add ToC to README

NicMcPhee committed:
Update LABTASKS

Mostly changes aimed at improving clarity and consistency.

I boldly replaced Protractor with Cypress everywhere, so now we "have" to make that change. :-)

helloworld12321 commented on LABTASKS.md:
```diff
@@ -27,7 +27,7 @@ All of the formatting (including the positioning of elements in the page) must
 - If you want to load Google Fonts you'll want to add those tags to your HTML so
   they'll be available in your CSS. (There are ways to do this entirely in CSS
-  but the "standard" approach typically involves modifying the HTML.)
+  but there are certain advantages do doing in the HTML when we start using Angular, so we'll do it that way here.)
```
```suggestion
  but there are certain advantages to doing it in the HTML when we start using Angular, so we'll do it that way here.)
```

Quick typo fix!

floogulinc commented on LABTASKS.md:
```diff
@@ -46,12 +46,12 @@ You should add a new section to the HTML (without changing any of the existing H
 
 - [`git`](https://git-scm.com/)
-- [GitHub](https://github.com/)
+- [GitHub.com](https://github.com/)
```
Why add ".com" here? I think it should be just called "GitHub".
```suggestion
- [GitHub](https://github.com/)
```

NicMcPhee replied:
Good question. It wasn't actually consistent before. I think it might have been written out as GitHub.com in a few places to distinguish that from github.umn.edu, which we used some for a while. We haven't used that in ages, though, so I could just change everything to GitHub.

floogulinc commented on LABTASKS.md:
```diff
@@ -63,9 +63,9 @@ Some requirements:
   Every other item should be in the class "local".
 - All items should be in a class indicating that tool's area of functionality:
-  - `version-control` for `git`, GitHub.com, and GitKraken`
+  - `version-control` for `git`, GitHub.com, and GitKraken
```
```suggestion
  - `version-control` for `git`, GitHub, and GitKraken
```

floogulinc commented on LABTASKS.md:
```diff
@@ -107,59 +111,47 @@ Please read carefully the list of technical requirements below and follow it pre
+    Safari, IE, Edge) and adapt the layout so that elements remain accessible and visible when the window is resized.
+    > To see how your page would look on some phones or tablets:
+    >
+    > Open your page in Chrome > right-click the page > click "Inspect" > click
```
Maybe we want to include that you can just hit `F12` or `ctrl/command + shift + i` for devtools.

helloworld12321 replied:
That'd certainly work too. I'd prefer to only give one set of instructions, though, just to avoid over-complicating things. 🙂

NicMcPhee replied:
`Command+shift+i` didn't work for me on Chrome on my Mac just now, although F12. This is the problem with these kinds of shortcuts; other software can block or interfere with them. I figure the kind of people who will be likely to remember and use a shortcut are also the kind of people who can figure out what the shortcuts are. 😜

floogulinc commented on README.md:
```diff
@@ -67,6 +84,21 @@ changes from feature branches into your `main` branch.
+
+Writing good commit messages:
+
+- [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) (from Chris Beams, 2014)
+- [How to write a good commit message](https://dev.to/chrissiemhrk/git-commit-message-5e21) (from Chrissie, 2020)
```
The second link here prescribes a pretty specific formula. I've personally never been a fan of the "type - subject" format described and I don't the second link really adds much coming from the first. This is really just my opinion though so feel free to ignore this comment entirely.

helloworld12321 replied:
Yep, seconded.

(Also, +1 for trimming down the number of articles the students have to read. 🙃)

floogulinc commented on README.md:
```diff
@@ -81,10 +113,9 @@ changes from feature branches into your `main` branch.
-- [FontJoy](http://fontjoy.com) – a cool tool for helping find combinations of fonts that work well together
+- [FontJoy](https://fontjoy.com/) - a nice tool for helping find combinations of fonts that work well together. (Google Fonts provides a similar feature as well.)
 - [Colormind](http://colormind.io) – a cool tool for helping find color combinations that work well together
 - [CSS Flexbox](https://www.w3schools.com/css/css3_flexbox.asp) - Add responsive design to automatically scale elements with page size
```
I've found myself using this resource frequently for flexbox: https://css-tricks.com/snippets/css/a-guide-to-flexbox/

Also this playground tool has been useful: https://demos.scotch.io/visual-guide-to-css3-flexbox-flexbox-playground/demos/ (there is something similar but specific to Angular we can include in future labs)

floogulinc commented on README.md:
```diff
@@ -81,10 +113,9 @@ changes from feature branches into your `main` branch.
 - [Colormind](http://colormind.io) – a cool tool for helping find color combinations that work well together
 - [CSS Flexbox](https://www.w3schools.com/css/css3_flexbox.asp) - Add responsive design to automatically scale elements with page size
+  - [A beginners guide to CSS flexbox - part one](https://dev.to/chrissiemhrk/a-beginners-guide-to-css-flexbox-part-one-29j1) (from Chrissie, 2020)
 - [CSS Media Queries](https://www.w3schools.com/css/css3_mediaqueries_ex.asp) - Add breakpoints to better design for specific sized screens
```
Maybe link to the general Media Queries page: https://www.w3schools.com/css/css3_mediaqueries.asp rather than the second "More Examples" page.

floogulinc commented on README.md:
```diff
@@ -81,10 +113,9 @@ changes from feature branches into your `main` branch.
   - [CSS Box Model](http://www.w3schools.com/css/css_boxmodel.asp)
   - [CSS Float](http://www.w3schools.com/css/css_float.asp)- How to position elements next to each other
   - [CSS Display and Visibility](http://www.w3schools.com/css/css_display_visibility.asp)
```
I'm not sure this one in particular is that useful for this lab.

helloworld12321 replied:
Float, you mean?

Yeah, I see what you're saying. I feel like floats are only really useful if you're trying to wrap text around an image or something. 🤷🏻‍♂️

floogulinc replied:
I did mean [CSS Display and Visibility](http://www.w3schools.com/css/css_display_visibility.asp) but maybe it is important to show the difference between block and inline?

NicMcPhee replied:
I think that all these CSS links have been added over the years in response to some specific student questions or confusions. The problem is that each person/group tends to go different directions and have different issues they wrestle with.

You don't want the list of resources to get infinitely long, reflecting a winding, complex past. On the other hand you don't want to remove things that have a high likelihood of being useful. I think that the difference between block and inline has come up numerous times, so I'm inclined to leave that bit in.

helloworld12321 replied:
Makes sense, thanks.

(FWIW, I'd also be inclined to keep Block vs. Inline—it's a decently big deal. 🙂)

floogulinc commented on README.md:
```diff
@@ -81,10 +113,9 @@ changes from feature branches into your `main` branch.
   - [CSS Box Model](http://www.w3schools.com/css/css_boxmodel.asp)
```
I can't directly comment on it because it's not quite within the changes but directly above this line there is the line:
```markdown
  - [A guide to font families](http://www.w3schools.com/cssref/css_websafe_fonts.asp)
```

This particular guide is kinda out of date as it doesn't include the concept of using [system fonts](https://css-tricks.com/snippets/css/system-font-stack/) which is a fairly common practice these days. (This is not a critical or blocking issue by any means, feel free to ignore.)

NicMcPhee replied:
Looking over the page you linked to, that adds significant complications, and still seems pretty fluid, where the "websafe fonts" will still work and in a generally predictable way. So I think I'm going to skip this for now. Feel free to extract it out into an issue, though, especially if there's a "simple" way to deal with this that doesn't involve listing a pile of system specific fonts.

NicMcPhee committed:
Apply suggestions from code review

Several suggested fixes from @helloworld12321 and @floogulinc.

Co-authored-by: Joe Moonan Walbran <walbr037@morris.umn.edu>
Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

NicMcPhee committed:
Address several suggestions re the README

@floogulinc and @helloworld12321 had a number of suggestions, especially regarding the list of resources. This incorporates many of those and tries to clarify the structure of the resources section at the end.

I also added a note about how GitKraken will help with some of the structure of a commit message.
